build_node/utils/git_utils.py

```diff
@@ -849,10 +853,16 @@ def __clone_repo(self, repo_url, target_dir, mirror=False, branch=None):
         @param target_dir:    The name of a new directory to clone into.
         @type mirror:         bool
         @param mirror:        Set up a mirror of the source repository if True.
+        @type git_opts:       list of str
+        @param git_opts:      list of explicit options to append to git command (optional)
 
         @raise GitCacheError: If git-clone execution failed.
         """
         cmd = ["git", "clone"]
+        if self.__git_command_extras is not None:
+            cmd.extend( self.__git_command_extras )
+        if git_opts is not None:
```

Comment:
What will happen if the option is duplicated in `self.__git_command_extras` and `git_opts`? Will git fail or just handle it?

Reply:
if you are adding options, `--all --all` for example, it'll take both and it's not an issue.  If someone puts something like `push` (or really anything that git isn't going to understand) in `git_extra_options` then it's going to cause a delightful mess as git errors all over the place, but I'm unsure if there's any real recourse against without having to fundamentally parse the entire command before it would get sent to git and attempt to throw an error before git (which doesn't make a whole lot of sense since git already has it's in-built parser, and the logs for a build are going to throw the error messages from git messing up as-is).

Reply:
Merged, thanks)